packages/babel-parser/src/parser/expression.js

```diff
@@ -852,6 +852,35 @@ export default class ExpressionParser extends LValParser {
     return this.parseSubscripts(this.parseExprAtom(), startPos, startLoc, true);
   }
 
+  /**
+   * Parse a primitive value
+   * primitive values are null, undefined, number, bigint, string, boolean
+   */
+  parsePrimitiveValue(): N.Expression {
+    let node;
+    switch (this.state.type) {
+      case tt.num:
+        return this.parseLiteral(this.state.value, "NumericLiteral");
+
+      case tt.bigint:
+        return this.parseLiteral(this.state.value, "BigIntLiteral");
+
+      case tt.string:
+        return this.parseLiteral(this.state.value, "StringLiteral");
+
+      case tt._null:
+        node = this.startNode();
+        this.next();
+        return this.finishNode(node, "NullLiteral");
+
+      case tt._true:
+      case tt._false:
+        return this.parseBooleanLiteral();
+      default:
```

Comment:
Nit: newline
```suggestion

      default:
```

Reply:
Have removed this function as we are only allowing String Literals. So marking this as resolved. Thank you 👍